### PR TITLE
tests.reporter.kubernetes minor cleanup

### PR DESCRIPTION
**Pull request by dankenigsberg**
I noticed a very odd [error log](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7044/pull-kubevirt-e2e-k8s-1.20-sig-compute-nonroot/1479758653413134336#1:build-log.txt%3A2835) where a [Cirros VM](https://github.com/kubevirt/kubevirt/blob/1b5960a/tests/vmi_configuration_test.go#L1061) is [not recognized as such](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7044/pull-kubevirt-e2e-k8s-1.20-sig-compute-nonroot/1479758653413134336#1:build-log.txt%3A2863).

I have no idea what is the problem there, but this PR would like to reduce the distraction caused by unrelated glitches.

```release-note
NONE
```

### CODE REVIEW

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -973,7 +973,7 @@ func (r *KubernetesReporter) dumpK8sEntityToFile(virtCli kubecli.KubevirtClient,
 
 	response, err := virtCli.RestClient().Get().RequestURI(requestURI).Do(context.Background()).Raw()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to dump entity named [%s]: %v\n", entityName, err)
+		fmt.Fprintf(os.Stderr, "dump %s: %v\n", entityName, err)
```
Comment (maya-r) [inline tests/reporter/kubernetes.go:976]: everything else looks good to me but I feel like this new text doesn't convey that it's a failure

Comment (dankenigsberg) [reply tests/reporter/kubernetes.go:976]: Well, my point is that in most of the cases (except of the very limited sriov lanes) this is *not* a failure. As it stands, it keeps distracting me when I search for `fail` in order to understand why a test lane failed, so I try to make it stand out less. Still, it is legible that something is amiss:
```
dump sriovoperatorconfigs: the server could not find the requested resource
```
Do you have a better text that does not use `fail`?

Ideally, each lane (or set of tests) would declare which CRDs they need to dump. Alas, we do not use pytest where doing this kind of things is trivial.

Comment (maya-r) [reply tests/reporter/kubernetes.go:976]: seeing what the text would be does make it clear it's a failure, thanks :-)<

File: tests/reporter/kubernetes.go [hunk 2]
```diff
@@ -1070,7 +1070,12 @@ func getVmiType(vmi v12.VirtualMachineInstance) (string, error) {
 		}
 	}
 
-	return "", fmt.Errorf("unknown type, vmi %s", vmi.ObjectMeta.Name)
+	vmiJson, err := json.Marshal(vmi)
+	if err == nil {
```
Comment (oshoval) [inline tests/reporter/kubernetes.go:1074]: what about `if vmiJson, err := json.Marshal(vmi); err == nil {` ?

Comment (dankenigsberg) [reply tests/reporter/kubernetes.go]: right, let's be go-idiomatic.

**Review by oshoval — APPROVED** [review]
Thanks

**Pull request merged**